discord/interactions/addConfigCommand.ts

```diff
@@ -280,34 +285,48 @@ export const handleConfigConfirmCommand = async (
 
   const currentConfig = ongoingConfigurationsCache[interaction.guildId];
 
+  const alreadyDiscordServerConfig =
+    await DiscordServerConfigRepository.findOneBy({
+      DiscordServerId: interaction.guildId,
+      starkyModuleConfig: currentConfig.moduleConfig,
+      discordRoleId: currentConfig.roleId,
+    });
```

Comment:
Shouldn't you add `starknetNetwork` to this search? Line 312 you may be REPLACING the `starknetNetwork` on an existing entity for a different network?